<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Classification of ALU Repeat Regions using Convolutional Neural Networks (CNNs)</title>
    <meta charset="utf-8" />
    <meta name="author" content="Dylan Loader" />
    <link rel="stylesheet" href="uofc.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: left, middle, title-slide

# Classification of ALU Repeat Regions using Convolutional Neural Networks (CNNs)
### Dylan Loader

---



layout: true
background-image: url("general-slide2.jpg")
background-size: cover

---

# Biological Sequences

![](transcription-translation.PNG)


---

# RNA Editing

.center[&lt;img src="r2i-editing.png" height="400" width="625"/&gt;]

???

Adenosine to Inosine editing. 




---
class: top
## ALU Transposons

A **Transposon** or **Transposable Element** is a DNA sequence which can change its position in the genome of a cell.

![Default-aligned image](Tranposons.png)

.footnote[Source:http://2014.igem.org/Team:Evry/Biology/Transposons]

???

-Also called 'jumping genes', Transposons are of interest in Genetics as they can cause mutations, cancers and changes to the genome size.

- Discovered in 
-Transposons can use a 'copy and paste' or 'cut and paste' method for transposition. 

-ALU transposons account for about 15%-17% of the human genome. 
---

# Data Description

- The data are obtained from [REDIportal](http://srv00.recas.ba.infn.it/atlas/). A database of 4.5 million A-to-I RNA editing events in 55 bodysites from 150 healthy human individuals. 

- Composed of RNA sequences of length 301 centered at an adenine. 

- There are three classes based on where the editing region is located:
  1. ALU repeat region     (n=50,000)
  2. Non-ALU repeat region (n=50,000)
  3. Non-repetitive region (n=50,000)
  
???
A-to-I is Adenine to Inosine(adapter molecule)

---

# Why CNNs?

&lt;img src="doggo.PNG" height="350" width="800"/&gt;

&lt;b&gt;Fasta File&lt;/b&gt;
&lt;img src="fasta.PNG"/&gt;

???

- CNNs predict especially well when spatial correlation exists, such as in genetic sequence data. 
- In problems with 'neat' and small data we usually use conventional methods. 
- messy data includes: rotations to the image, rescaling, and distortion. 
---
class: left

# pysster: a Sequence-STructure classifiER

.pull-left[&lt;img src="cnn-diagram.png"/&gt;]
.pull-right[&lt;img src="my-cnn.PNG"/&gt;]

???

- 3 convolution and pooling layers (ReLU function)
- 2 dense layers (we only care about the last layer for weights)
- Use a soft max for the three class, classification
---

# Validation Loss

.center[&lt;img src="Rplot02.png"  height="500"/&gt;]

---
# Model Summary

.center[&lt;img src="model-acc.PNG" &lt;img src="perf-roc2.png"  height="525" /&gt;]

???
- One vs. all classifiction to predict ROC
- Precision = `\(TP/(TP+FP)\)`
- Recall = `\(TP/(TP+FN)\)`
- F1 = `\(\frac{2*recall*precision}{Recall+precision}\)`, penalizes extreme values caused by
---
# What else can this package do?


1. Recurrent Neural Networks (RNNs) and Long short-term memory (LSTM)

2. Visual expression for DNA and protein motifs:

&lt;img src="motif.PNG"  height="250" width="2375"/&gt;

---

# Other areas using Deep Learning in Bioinformatics


AlphaFold by Google's DeepMind team: 
&lt;img src="alpha.gif"/&gt;

.footnote[https://deepmind.com/blog/alphafold/]
---

# Questions?

**References:**

- "pysster: classification of biological sequences by learning sequence and structure motifs with convolutional neural networks."
Budach, Marsico.
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"navigation": {
"click = TRUE": null
},
"countIncrementalSlides": false,
"highlightLines": true,
"highlightStyle": "github"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
